### PR TITLE
fix: should raise exception when apply a categorical axis

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

Raise a specified error when using categorical x-axis and time comparison.

#### After
image

#### Before
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by stephenLYZ — APPROVED** [review]
LGTM

File: superset/common/query_context_processor.py [hunk 1]
```diff
@@ -322,6 +322,16 @@ def processing_time_offsets(  # pylint: disable=too-many-locals
                 #  multi-dimensional charts
                 granularity = query_object.granularity
                 index = granularity if granularity in df.columns else DTTM_ALIAS
+                if not pd.api.types.is_datetime64_any_dtype(
```
Comment (mistercrunch) [inline superset/common/query_context_processor.py:325]: Based on recent errors I'm guessing this is too restrictive.

Comment (mistercrunch) [reply superset/common/query_context_processor.py:325]: Guessing this should be `is_datetime64_any_dtype(...) or is_datetime64tz_any_dtype(...)`

Comment (mistercrunch) [reply superset/common/query_context_processor.py:325]: Screen Shot 2022-07-05 at 12 32 20 PM

Comment (zhaoyongjie) [reply superset/common/query_context_processor.py:325]: I will open a new PR and depend on the Generic Type of column instead of Pandas type.

Comment (zhaoyongjie) [reply superset/common/query_context_processor.py:325]: `is_datetime64_any_dtype` verify `is_datetime64tz_any_dtype` as well, https://github.com/pandas-dev/pandas/blob/e8093ba372f9adfe79439d90fe74b0b5b6dea9d6/pandas/core/dtypes/common.py#L922

**Pull request merged**